bigdecimal/src/lib.rs

```diff
@@ -0,0 +1,637 @@
+        }
+
+        let BIG_TEN = &BigInt::from_i8(10).unwrap();
+        let mut remainder = remainder * BIG_TEN;
+        let mut quotient = quotient;
+
+        let MAX_ITERATIONS = 100;
+        let mut iteration_count = 0;
+        while remainder != BigInt::zero() && iteration_count < MAX_ITERATIONS {
+            let (q, r) = remainder.div_rem(&den);
+            quotient = quotient * BIG_TEN + q;
+            remainder = r * BIG_TEN;
+
+            iteration_count += 1;
+        }
+        let scale = scale + iteration_count;
+        BigDecimal::new(quotient, scale)
+    }
+}
+
+forward_all_binop_to_ref_ref!(impl Rem for BigDecimal, rem);
+
+impl<'a, 'b> Rem<&'b BigDecimal> for &'a BigDecimal {
+    type Output = BigDecimal;
+
+    #[inline]
+    fn rem(self, _: &BigDecimal) -> BigDecimal {
+        // let (_, r) = self.div_rem(other);
+        // return r;
+        self.clone()
+    }
+}
+
+impl fmt::Display for BigDecimal {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.pad_integral(true, "", &self.int_val.to_str_radix(10))
+    }
+}
+
+impl Num for BigDecimal {
+    type FromStrRadixErr = ParseBigDecimalError;
+
+    /// Creates and initializes a BigDecimal.
+    #[inline]
+    fn from_str_radix(s: &str, radix: u32) -> Result<BigDecimal, ParseBigDecimalError> {
+        assert!(2 <= radix && radix <= 36, "The radix must be within 2...36");
+
+        let exp_separator: &[_] = &['e', 'E'];
+
+        // split slice into base and exponent parts
+        let (base_part, exponent_value) = match s.find(exp_separator) {
+            // exponent defaults to 0 if (e|E) not found
+            None => (s, 0),
+
+            // split and parse exponent field
+            Some(loc) => {
+                let (base, exp) = s.split_at(loc);
+                // slice after 1 to skip the 'e' char
+                (base, try!(i64::from_str(&exp[1..])))
+            }
+        };
+
+        // TEMPORARY: Test for emptiness - remove once BigInt supports similar error
+        if base_part == "" {
+            return Err(ParseBigDecimalError::Empty);
+        }
+
+        // split decimal into a digit string and decimal-point offset
+        let (digits, decimal_offset) = match base_part.find('.') {
+            // No dot! pass directly to BigInt
+            None => (base_part.to_string(), 0),
+
+            // decimal point found - necessary copy into new string buffer
+            Some(loc) => {
+                // split into leading and trailing digits
+                let (lead, trail) = base_part.split_at(loc);
+
+                // copy all leading characters into 'digits' string
+                let mut digits = String::from_str(lead).unwrap();
+
+                // copy all trailing characters after '.' into the digits string
+                digits.extend(trail.chars().skip(1));
+
+                (digits, trail.len() as i64 - 1)
+            }
+        };
+
+        let scale = decimal_offset - exponent_value;
```

Comment:
Hmm, actually on the radix point, this is wrong for radixes other than 10, and others can't be perfectly converted.
(e.g. `"0.1"` radix 3 is `1/3`)